www/routes/proxy-route.ts

```diff
@@ -0,0 +1,58 @@
+import type { HTTPMiddleware } from "revolution";
+import { call, Operation } from "effection";
+
+export interface ProxyRouteOptions {
+  website: string;
+  prefix: string;
+  root?: string;
+}
+
+export function proxyRoute(options: ProxyRouteOptions): HTTPMiddleware {
+  return function* proxy(request): Operation<Response> {
+    let website = new URL(options.website);
+
+    let target = new URL(request.url);
+
+    let prefix = new RegExp(`^\/${options.prefix}\/?`);
+    target.pathname = target.pathname.replace(prefix, options.root ?? "/");
+
+    target.hostname = website.hostname;
+    target.port = website.port;
+    target.protocol = website.protocol;
+
+    let base = new URL(`/${options.prefix}`, request.url);
+
+    let headers: Record<string, string> = {
+      "X-Base-Url": base.toString(),
+    };
```

Comment:
Are we using this base header in the contrib website?

I think we may want to start statically generating the effection website at this point. because it's happening in the context of us also needing to generate the frontside website as well. Let's pair up to get that done